### PR TITLE
Add initial foundations

### PR DESCRIPTION
**Pull request by illusional**
Adds initial foundations for sample metadata system:

- A DB schema, applied using `liquibase`
- A set of DB classes + models
    - Tables + Layers that can interact with a MySQL database
    - If `export SM_ENVIRONMENT="PRODUCTION"`, then it will pull credentials from `projects/sample_metadata/secrets/databases/`, maintains one connection per database.
- A web API with Flask + `apispec`:
    - uses OpenAPI3 to generate a swagger UI, and an installable `sample_metadata` python API.
    - Authentication of the web API using the GCP auth tokens.
    - Currently NO authorization - except the web API will be deployed behind an Google Identity-aware proxy (IAP)
    - Capturing the current user from this bearer token
- Installable API
    - Generated from OpenAPI3 annotations, using the [openapi-generator](https://openapi-generator.tech/docs/installation/). 
    - `regenerate_api.py` is a script that can generate the installable sample_metadata API. Can use an existing instance, or spin up an instance (including using Docker)

### CODE REVIEW

**Review by vivbak — APPROVED** [review]
:)

**Comment by vladsaveliev** [comment]
Wanted to push some updates to the README too as my setup differed a bit. But figured I don't want to mess with your README, as my setup might not be ideal. Can have chat on zoom on how to organize README to include my setup, if it makes sense? 

Here it is:


### Using Maria DB docker image

Pull mariadb image

```bash
docker pull mariadb
```

Run a mariadb container that will server your database. `-p 3307:3306` remaps the port to 3307 in case if you local MySQL is already using 3306

```bash
docker stop mysql-p3307  # stop and remove if the container already exists
docker rm mysql-p3307
docker run -p 3307:3306 --name mysql-p3307 -e MYSQL_ROOT_PASSWORD=root -d mariadb
```

Initialize databases (you may need to enter the password on each command).

```bash
mysql --host=127.0.0.1 --port=3307 -u root -p -e 'CREATE DATABASE sm_dev;'
mysql --host=127.0.0.1 --port=3307 -u root -p -e 'CREATE DATABASE sm_admin;'
mysql --host=127.0.0.1 --port=3307 -u root -p -e 'show databases;'
```

Download the mariadb jar into the `db/` folder from https://downloads.mariadb.org/connector-java/2.7.3.

Then create the database schemas.

```bash
pushd db/
liquibase update --url jdbc:mariadb://127.0.0.1:3307/sm_admin --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changelog-file=global.xml
liquibase update --url jdbc:mariadb://127.0.0.1:3307/sm_dev --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changelog-file=project.xml
```

Finally, start the server, making use of the environment variables to point it to your local Maria DB server

```bash
export SM_DEV_DB_PROJECT=sm_dev
export SM_DEV_DB_USER=root
export SM_DEV_DB_PORT=3307
export SM_DEV_DB_HOST=127.0.0.1
export SM_DEV_DB_PASSWORD=root
export SM_ENVIRONMENT=dev
export SM_USE_SERVICE_ACCOUNT=false
python3 -m api.server
```

**Comment by illusional** [comment]
Thanks @vladsaveliev, this is awesome! I've incorporated your changes here, and slightly refreshed the README: https://github.com/populationgenomics/sample-metadata/blob/add-foundations/README.md

Does this look okay?

**Comment by vladsaveliev** [comment]
That's awesome!

Just one q, did `wget https://cdn.mysql.com//Downloads/Connector-J/mysql-connector-java-8.0.24.zip` get you `mariadb-java-client-*.jar`? For me, I had to download it separately, but probably I'm missing something.

**Comment by illusional** [comment]
@vladsaveliev, I've been using the mariadb connector from here: https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.2/mariadb-java-client-2.7.2.jar. I realise this is different to BOTH links I've posted, I've update the README.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,210 @@
...
+mysql --host=127.0.0.1 --port=3307 -u root -p -e 'show databases;'
+```
+
+Download [`mariadb-java-client-2.7.2.jar`](https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.2/mariadb-java-client-2.7.2.jar) into the `db/` folder.
+
+Then create the database schemas.
+
+```bash
+pushd db/
+liquibase update --url jdbc:mariadb://127.0.0.1:3307/sm_admin --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changelog-file=global.xml
```
Comment (vivbak) [inline README.md:95]: I think this is supposed to be 
`liquibase --url jdbc:mariadb://127.0.0.1:3307/sm_admin --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changeLogFile global.xml update`

Comment (illusional) [reply README.md]: Thanks, I've fixed this in the latest commit.

**Pull request merged**